### PR TITLE
Fix using symbol on Element#attribute alwas getting nil

### LINKED ISSUE #173: Element#attribute does not return correct result with Symbol input
In [doc](http://code.yorickpeterse.com/oga/latest/Oga/XML/Element.html#attribute-instance_method)
The input mentioned supporting `String` AND `Symbol`

But by testing and looking through the source code
It never returns the attribute if symbol is passed to the method

The internal method [`attribute_matches?`](https://github.com/YorickPeterse/oga/blob/master/lib/oga/xml/element.rb#L350) does not convert symbols to strings
And neither the [`attribute` method](https://github.com/YorickPeterse/oga/blob/master/lib/oga/xml/element.rb#L66)

### PR DESCRIPTION
**Pull request by PikachuEXE**
Fixes #173 

I also refactor `split_name` (change var name only)
If you don't want that change to be included let me know

### CODE REVIEW

**Comment by PikachuEXE** [comment]
AppVeyor failed due to nokogiri installation issue, not related to spec

File: spec/oga/xml/element_spec.rb [hunk 1]
```diff
@@ -107,13 +107,23 @@
...
-        doc  = Oga::XML::Document.new(:children => [el], :type => :html)
+      let!(:attr) do
+        Oga::XML::Attribute.new(:name => 'foo:bar', :value => 'foo')
+      end
+      let!(:el) do
+        described_class.new(:name => 'foo', :attributes => [attr])
+      end
+      let!(:doc) do
+        Oga::XML::Document.new(:children => [el], :type => :html)
+      end
```
Comment (YorickPeterse) [inline spec/oga/xml/element_spec.rb]:
```ruby
let(:attr) do
  Oga::XML::Attribute.new(:name => 'foo:bar', :value => 'foo')
end

let(:el) do
  el = described_class.new(:name => 'foo', :attributes => [attr])
  doc = Oga::XML::Document.new(:children => [el], :type => :html)

  el
end
```

This removes the need for `let!`, and the let binding for `doc`.

**Comment by YorickPeterse** [comment]
@PikachuEXE Could you also add a changelog entry in similar style to the previous ones? You can set the release date to "Unreleased" for now.

**Comment by PikachuEXE** [comment]
@YorickPeterse 
Ok I updated the CHANGELOG
If you think the format is incorrect just fix after merging it :P

**Comment by YorickPeterse** [comment]
This has been released in version 2.10.

**Pull request merged**